core/src/com/unciv/ui/screens/worldscreen/unit/actions/UnitActionModifiers.kt

```diff
@@ -63,13 +63,27 @@ object UnitActionModifiers {
         return true
     }
 
+    private fun canSpendStockpileCost(unit: MapUnit, actionUnique: Unique): Boolean {
+        for (conditional in actionUnique.conditionals.filter { it.type == UniqueType.UnitActionStockpileCost }) {
+            val amount = conditional.params[0].toInt()
+            val resourceName = conditional.params[1]
+            if (unit.civ.getCivResourcesByName()[resourceName]!! < amount) {
```

Comment:
A malformed resourceName will cause crash, I think we have a safe version? If not then do like 'val amount = ... ?: return false'

Reply:
Found the safety version